includes/Core/Admin_Bar/Admin_Bar.php

```diff
@@ -92,6 +92,20 @@ function( $wp_admin_bar ) {
 		add_action( 'wp_enqueue_scripts', $admin_bar_callback, 40 );
 	}
 
+	/**
+	 * Add data-ampdevmode attributes to the elements that need it.
+	 *
+	 * @see \Google\Site_Kit\Core\Assets\Assets::get_assets() The 'googlesitekit' string is added to all inline scripts.
+	 * @see \Google\Site_Kit\Core\Assets\Assets::add_amp_dev_mode_attributes() The data-ampdevmode attribute is added to registered scripts/styles here.
+	 *
+	 * @param string[] $xpath_queries XPath queries for elements that should get the data-ampdevmode attribute.
+	 * @return string[] XPath queries.
+	 */
+	public function add_amp_dev_mode( $xpath_queries ) {
+		$xpath_queries[] = '//script[ contains( text(), "googlesitekit" ) ]';
```

Comment:
Just to clarify, this is only for inline scripts then right? The non-inline scripts receive the attribute in their own `*_loader_tag` hooks I see.

Adding a random `googlesitekit` comment to all these scripts is not so great. Is there not a way to add the `data-ampdevmode` attribute to those inline scripts with a filter too?

Reply:
> Just to clarify, this is only for inline scripts then right? The non-inline scripts receive the attribute in their own `*_loader_tag` hooks I see.

Yes, that is correct.

> Adding a random `googlesitekit` comment to all these scripts is not so great. Is there not a way to add the `data-ampdevmode` attribute to those inline scripts with a filter too?

It's not awesome, but it is effective and somewhat nice for debugging to locate and identify scripts coming from Site Kit.

I did try to find a way to filter the inline scripts to inject this `data-amp-dev-mode` attribute, but I was unable to. The extra scripts get printed in `\WP_Scripts::print_extra_script()` which unfortunately has no filter like `script_loader_tag`.